src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -87,8 +86,11 @@ CsvImportWidget::CsvImportWidget(QWidget* parent)
         combo->setFont(font);
         m_combos.append(combo);
         combo->setModel(m_comboModel);
-        m_comboMapper->setMapping(combo, i);
-        connect(combo, SIGNAL(currentIndexChanged(int)), m_comboMapper, SLOT(map()));
+#if QT_VERSION >= QT_VERSION_CHECK(5, 7, 0)
+        connect(combo, QOverload<int>::of(&QComboBox::currentIndexChanged), [=]{ comboChanged(combo, i); });
+#else
+        connect(combo, static_cast<void(QComboBox::*)(int)>(&QComboBox::currentIndexChanged), [=]{ comboChanged(combo, i); });
+#endif
```

Comment:
This is very ugly, almost incomprehensible. I feel like this interaction is just not coded well to begin with due to the use of the QSignalMapper in the first place.

Reply:
To be fair I don't love it either, but there's no other way I know of I can tell the compiler to pick
```cpp
void currentIndexChanged(int)
```
instead of
```cpp
void currentIndexChanged(const QString&)
```

The `QOverload<int>::of()` syntax is a lot more readable in my opinion but it's only available since Qt 5.7.

Reply:
Actually, the AppImage build uses a more recent Qt version (5.10 I believe).